platforms/documentation/docs/src/docs/userguide/dep-man/dependency-management/resolution_rules.adoc

```diff
@@ -0,0 +1,474 @@
+- The substitution replaces the module dependency with the project dependency and sets up task dependencies, but doesn't automatically include the project in the build.
+
+[[sub:project_to_module_substitution]]
+=== Substituting a project dependency with a module replacement
+
+You can also use substitution rules to replace a project dependency with an external module in a multi-project build.
+
+This technique can accelerate development by allowing certain dependencies to be downloaded from a repository instead of being built locally:
+
+====
+include::sample[dir="snippets/dependencyManagement/customizingResolution-substitutionRule/kotlin",files="build.gradle.kts[tags=project_to_module_substitution]"]
+include::sample[dir="snippets/dependencyManagement/customizingResolution-substitutionRule/groovy",files="build.gradle[tags=project_to_module_substitution]"]
+====
+
+- The substituted module must include a version.
+- Even after substitution, the project remains part of the multi-project build, but tasks to build it won't be executed when resolving the configuration.
+
+=== Forcing a dependency version
+
+You can enforce a specific version of a dependency, regardless of what versions might be requested or resolved by other parts of the build script.
+
+This is useful for ensuring consistency and avoiding conflicts due to different versions of the same dependency being used.
+
+====
+include::sample[dir="snippets/dependencyManagement/managingTransitiveDependencies-forceForConfiguration/kotlin",files="build.gradle.kts[tags=force-per-configuration]"]
+include::sample[dir="snippets/dependencyManagement/managingTransitiveDependencies-forceForConfiguration/groovy",files="build.gradle[tags=force-per-configuration]"]
+====
+
+[[sub:conditional_dependency_substitution]]
+=== Conditionally substituting a dependency
+
+You can conditionally substitute a module dependency with a local project in a multi-project build using dependency substitution rules.
+
+This is particularly useful when you want to use a locally developed version of a dependency if it exists, otherwise fall back to the external module:
+
+====
+include::sample[dir="snippets/dependencyManagement/customizingResolution-conditionalSubstitutionRule/kotlin/consumer",files="build.gradle.kts[tags=project_substitution]"]
+include::sample[dir="snippets/dependencyManagement/customizingResolution-conditionalSubstitutionRule/groovy/consumer",files="build.gradle[tags=project_substitution]"]
+====
+
+- The substitution only occurs if a local project matching the dependency name is found.
+- The local project must already be included in the multi-project build (via `settings.gradle`).
+
+[[sec:variant_aware_substitutions]]
+=== Substituting a dependency with another variant
+
+you can substitute a dependency with another <<variant_model.adoc#understanding-variant-selection,variant>>, such as switching between a platform dependency and a regular library dependency.
```

Comment:
```suggestion
You can substitute a dependency with another <<variant_model.adoc#understanding-variant-selection,variant>>, such as switching between a platform dependency and a regular library dependency.
```

Reply:
💭 Also, I think this section is unclear.  What is meant by a platform dependency?

Reply:
Fixed.